8.18.0 Release notes (#6519)
elastic/security-docs: docs/release-notes/8.18.asciidoc

//* adds upgrade notes and create docs link for Endpoint management deprecated apis in 9.0 ({kibana-pull}206903[#206903]).
//* Adds deprecation warning for the legacy risk score modules ({kibana-pull}202775[#202775]) for details.
//Might need to elaborate on the following summary and also doc it in the Kibana release notes at https://www.elastic.co/guide/en/kibana/8.18/release-notes-8.18.0.html.
* The following SIEM signal migration endpoints were deprecated ({kibana-pull}202662[#202662]):

** POST /api/detection_engine/signals/migrations
** DELETE /api/detection_engine/signals/migrations
** POST /api/detection_engine/signals/finalize_migrations
** GET /api/detection_engine/signals/migration_status

[discrete]
[[features-8.18.0]]
==== New features
* The Automatic Import functionality is now generally available ({kibana-pull}208523[#208523]).
* Allows you to install and reinstall entity stores from the Engine Status page ({kibana-pull}208149[#208149]).
* Adds enrichPolicyExecutionInterval to entity enablement and init APIs ({kibana-pull}207374[#207374], {kibana-pull}204437[#204437]).

> * Provides initial support for the service entity type, whereas previously, only user and host entity types were supported ({kibana-pull}207336[#207336], {kibana-pull}206582[#206582], {kibana-pull}206268[#206268], {kibana-pull}202344[#202344]).

Reviewer: I want to mention that we are adding the service type to the **Entity Store**.
```suggestion
* Adds initial support for the service entity type in the Entity Store, whereas previously, only user and host entity types were supported ({kibana-pull}207336[#207336], {kibana-pull}206582[#206582], {kibana-pull}206268[#206268], {kibana-pull}202344[#202344]).
```

* Adds audit logging for changes to AI Assistant knowledge base entries ({kibana-pull}203349[#203349]).
* Adds a service example to the entity store upload page ({kibana-pull}209023[#209023]).
* Update entity insight badge to open entity flyouts ({kibana-pull}208287[#208287]).
* Introduces changes to the entity analytics feature to support `event.ingested` as a configurable timestamp field for init and enable endpoints ({kibana-pull}208201[#208201]).
* Allows you to include closed alerts in risk score calculations ({kibana-pull}201909[#201909]).
* Turns the `securitySolution:enableVisualizationsInFlyout` <<visualizations-in-flyout,advanced setting>> on by default, which allows you to access the event analyzer and Session View in the **Visualize** tab on the alert or event details flyout ({kibana-pull}211319[#211319]).
* Reduces the system performance impact of {elastic-defend} file events.
* Improves {elastic-defend}'s resilience in low memory situations.
* Updates the {elastic-defend} policy status message to show the {elastic-defend} policy name, revision, and {agent} policy revision.

> * Ensures that the data view selector on the rule creation form shows data view names instead of their defined indices ({kibana-pull}214495[#214495]).

Author: @approksiu and @yctercero made this change in response to https://github.com/elastic/security-docs/pull/6519#issuecomment-2759463443. The PR is labeled to be skipped for release notes, so let me know if I should link to a different one. Thanks! 

```suggestion
* Ensures that the data view selector on the rule creation form shows data view names instead of their defined indices ({kibana-pull}214495[#214495]).
* Allows rule actions (except for **Summary of alerts** actions that run at a custom frequency) to activate during manual rule runs ({kibana-pull}208072[#208072]).
```

Reviewer 2: Here is the original PR: https://github.com/elastic/kibana/pull/200784, strange why it's skip notes, it's an important improvement.

Author: ```suggestion
* Ensures that the data view selector on the rule creation form shows data view names instead of their defined indices ({kibana-pull}214495[#214495]).
* Allows rule actions (except for **Summary of alerts** actions that run at a custom frequency) to activate during manual rule runs ({kibana-pull}200784[#200784]).
```

Reviewer 3: Agree it should be release-noted 👍🏽

* Fixes the order of the alert insights so they're now shown from low risk to critical risk({kibana-pull}212980[#212980]).
* Fixes bugs that prevents cell action in the Alerts table from properly rendering in the event rendered view ({kibana-pull}212721[#212721]).
* Fixes a bug that incorrectly concealed the the isolate host panel if you used the isolate host action from the alert preview ({kibana-pull}211853[#211853]).
* Fixes a bug that prevented you from seeing alert assignee details from the Alerts table or the alert details flyout ({kibana-pull}211824[#211824]).
* Fixes the width of the alerts table in rule preview ({kibana-pull}214028[#214028]).
* Fixes a bug that prevented the rule creation form from properly validating EQL queries when you added filters to the query ({kibana-pull}212117[#212117]).
* Makes 7.x alert indices compatible with Alerts table so you can access alerts in legacy indices ({kibana-pull}209936[#209936]).
* Fixes a bug that didn't allow you to generate {esql} alerts from alert indices ({kibana-pull}208894[#208894]).

> * Surfaces details for failed EQL non-sequence queries on the rule details page and in the event log ({kibana-pull}207396[#207396]).

Author: From Slack convo with Devin: 

```suggestion
* Surfaces shard failure details for failed EQL non-sequence queries on the rule details page and in the event log ({kibana-pull}207396[#207396]).
```

* Fixes an {elastic-defend} bug to ensure the first event's timestamp is used as the timestamp for event aggregation.
* Updates the way {elastic-defend} initially connects to {agent}, which significantly improves the speed of connection.

> * Fixes issues where uninstalling {elastic-defend] on Windows leaves files within {elastic-defend}'s directory that cannot be removed by administrators. These leftover files can prevent subsequent installs and upgrades.

Reviewer 4: {elastic-defend] -> {elastic-defend}

Author: Addressed in [84b01ce](https://github.com/elastic/security-docs/pull/6519/commits/84b01cef0162353ff08baf9206adedacce200711).

* Allows you to https://github.com/elastic/kibana/issues/174168[customize prebuilt rules]. You can modify most rule parameters, export and import prebuilt rules — including customized ones — and upgrade prebuilt rules while retaining customization settings ({kibana-pull}212761[#212761]).
* Adds initial support for the service entity type in the Entity Store, whereas previously, only user and host entity types were supported ({kibana-pull}207336[#207336], {kibana-pull}206582[#206582], {kibana-pull}206268[#206268], {kibana-pull}202344[#202344]).
* Allows you to configure how often the enrich policy runs for the entity store ({kibana-pull}207374[#207374], {kibana-pull}204437[#204437]).
* Provides configuration options to the entity store through additional API parameters ({kibana-pull}206421[#206421]).
* Introduces a status tab to the entity store management page ({kibana-pull}201235[#201235]).
* Allows you to install and reinstall entity stores from the Engine Status page ({kibana-pull}208149[#208149]).

> * Introduces ways to monitor and fix gaps in rule executions, which can lead to missed alerts or reduced rule coverage ({kibana-pull}206313[#206313]).

Author: @nkhristinin @approksiu do you happen to know the PR that made manual runs GA?

```suggestion
* Introduces ways to monitor and fix gaps in rule executions, which can lead to missed alerts or reduced rule coverage ({kibana-pull}206313[#206313]).
* The manual runs functionality is now generally available ({kibana-pull}209535[#209535]).
```

Reviewer 3: Here is the PR - https://github.com/elastic/kibana/pull/209535

Author: Thanks! Added ^

[discrete]
[[bug-fixes-8.18.0]]
==== Bug fixes
* Fixes the unstructured system log flow for Automatic Import ({kibana-pull}213042[#213042]).
* Fixes missing ECS mappings for Automatic Import ({kibana-pull}209057[#209057]).
* Fixes how Automatic Import generates accesses for the field names that are not valid Painless identifiers ({kibana-pull}205220[#205220]).
* Automatic Import now ensures that the field mapping contains the `@timestamp` field whenever possible ({kibana-pull}204931[#204931]).
* Ensures that Automatic Import uses the provided data stream description in the integration readme ({kibana-pull}203236[#203236]).
* Fixes the countdown for the next scheduled risk engine run ({kibana-pull}203212[#203212]).
* Ensures that Automatic Import uses the data stream name that you provide instead of a generic placeholder ({kibana-pull}203106[#203106]).
* Fixes the bug where pressing Enter reloaded the Automatic Import ({kibana-pull}199894[#199894]).
* Fixes a bug that prevented you from being able to select a connector for AI Assistant from the {elastic-sec} landing page ({kibana-pull}213969[#213969]).

> * Updates prompts that you can use with the Amazon Bedrock connector ({kibana-pull}213160[#213160]).

Reviewer 5: ```suggestion
* Updates prompts that you can use with the Amazon Bedrock connector ({kibana-pull}213160[#213160]).
* Fixes a bug in AI Assistant that caused the Bedrock region to always be `us-east-1` ({kibana-pull}213160[#214251]).
```

One more bug fix we should call out here

* Allows you to add common observables to any case and extend the types of observable case data to include custom options ({kibana-pull}190237[#190237]).
* Introduces privileges that let you control role access to Timeline and notes ({kibana-pull}201780[#201780]).
* Introduces privileges that let you control whether a role can assign users to a case ({kibana-pull}201654[#201654]).
* Re-adds details to the alert details flyout about the last time an alert's status was changed ({kibana-pull}205224[#205224]).
* Introduces changes to the asset criticality and risk score data clients to use a new ingest pipeline for adding event timestamps ({kibana-pull}203975[#203975]).
* Adds new third-party actions to CrowdStrike response actions, which will allow you to execute remote commands using Crowdstrike agent through {elastic-sec} ({kibana-pull}203101[#203101], {kibana-pull}202012[#202012], {kibana-pull}203420[#203420], {kibana-pull}204044[#204044]).
* Applies the latest Elastic UI (EUI) theme to multiple areas of {elastic-sec} ({kibana-pull}204007[#204007], {kibana-pull}204908[#204908]).

> * Adds the {elastic-defend} `[os].advanced.artifacts.global.channel` <<adv-policy-settings,advanced policy setting>>, which allows you to opt out from staged artifact rollout ({kibana-pull}202674[#202674]).

Reviewer 6: ```suggestion
```

See https://github.com/elastic/security-docs/pull/6519/files#r2014369698

> * Adds new {elastic-defend} fields, `process.Ext.command_line_truncated` and `process.parent.Ext.command_line_truncated` to indicate when the command line gathered by event sources is truncated because of size limitations.

Reviewer 6: ```suggestion
*  {elastic-defend} will now graphically report its protection status when launched from Windows Security Center.
* Adds new {elastic-defend} fields, `process.Ext.command_line_truncated` and `process.parent.Ext.command_line_truncated` to indicate when the command line gathered by event sources is truncated because of size limitations.
```

* Allows {elastic-defend} to send data to telemetry.elastic.co to monitor the health of staged global artifacts rollouts.
* Updates the infrastructure of HttpClient to allow for future implementation of a Rust based client.
* Adds infrastructure to CryptoLib that will enable a smoother transition to a Rust CryptoLib implementation.

> * Ensures that global artifacts update are delivered incrementally, closely monitoring the health of the rollout. To support it, {elastic-defend} will contact a new cloud API to know which artifacts it should use, will contact Elastic telemetry to send periodic health information during artifacts testing, and lastly, will allow you to use the advanced setting in your {elastic-defend} policy to opt-out from participating in the staged artifacts rollout.

Reviewer 6: Here's some combined verbiage to use as a starting point.

```suggestion
* {elastic-defend} staged artifact rollout is now generally available.  Staged artifact rollout incrementally updates global artifacts including malware models and behavioral rules.  Each update cycle begins with a small percentage of cloud-connected systems receiving new artifacts.  These systems then report any stability, performance, and protection efficacy issues.  Over time, additional systems will receive the updates until all systems are updated to the latest artifacts.  If any issues are identified, Elastic may halt the update process and rollback all participating systems to prior known-good artifacts.  To support this process, participating {elastic-defend} endpoints will report health-related telemetry to `telemetry.elastic.co`.  Customers can control this behavior via the `[os].advanced.artifacts.global.channel` <<adv-policy-settings,advanced policy setting>> ({kibana-pull}202674[#202674]).
```

> * Ensures that global artifacts update are delivered incrementally, closely monitoring the health of the rollout. To support it, {elastic-defend} will contact a new cloud API to know which artifacts it should use, will contact Elastic telemetry to send periodic health information during artifacts testing, and lastly, will allow you to use the advanced setting in your {elastic-defend} policy to opt-out from participating in the staged artifacts rollout.

Author: Light edits for punctuation and to remove extra spaces.
```suggestion
* {elastic-defend} staged artifact rollout is now generally available. Staged artifact rollout incrementally updates global artifacts, including malware models and behavioral rules. Each update cycle begins with a small percentage of cloud-connected systems receiving new artifacts. These systems then report any stability, performance, and protection efficacy issues. Over time, additional systems will receive the updates until all systems are updated to the latest artifacts. If any issues are identified, Elastic may halt the update process and rollback all participating systems to prior known-good artifacts. To support this process, participating {elastic-defend} endpoints will report health-related telemetry to `telemetry.elastic.co`. Customers can control this behavior using the `[os].advanced.artifacts.global.channel` <<adv-policy-settings,advanced policy setting>> ({kibana-pull}202674[#202674]).
```

* Enables process event aggregation by default.

> * Improves {elastic-defend} by adding inherited event counting (events from children) to the process cache entry.

Reviewer 7: Yes, my bad.  I should have put `release_note: skip` on that

* Adds a new field to the metrics section of the metadata document called `top_process_trees`. This section will contain a list of the top noisy processes on the system, with "noisy" being based on how many events they generate.

> * Allows you to opt into aggregation of network events to reduce CPU usage, I/O, and event sizes. Network events with the same addresses and ports occurring in rapid succession will be combined into fewer aggregate events. Use the `advanced.events.aggregate_network` advanced setting in your {elastic-defend} policy to enable it.

Author: Yeah, that'd work. @ferullo if you don't mind drafting a release notes summary of the changes, that'd be a good starting point.

> * Allows you to opt into aggregation of network events to reduce CPU usage, I/O, and event sizes. Network events with the same addresses and ports occurring in rapid succession will be combined into fewer aggregate events. Use the `advanced.events.aggregate_network` advanced setting in your {elastic-defend} policy to enable it.

Reviewer 8: How about this? I'm not sure if you want to link to this documentation somewhere in there? https://www.elastic.co/guide/en/security/current/endpoint-data-volume.html

> To reduce data volume, by default {elastic-defend} will now merge short lived process create/terminate events and network connect/terminate events so only a single document is produced, the host fieldset will contain less fields in event documents, and MD5 and SHA1 hashes will be disabled. All existing {elastic-defend} policies will preserve the old behavior. All new behaviors can be disabled using advanced settings.

> * Allows you to opt into aggregation of network events to reduce CPU usage, I/O, and event sizes. Network events with the same addresses and ports occurring in rapid succession will be combined into fewer aggregate events. Use the `advanced.events.aggregate_network` advanced setting in your {elastic-defend} policy to enable it.

Author: Thanks, @ferullo. I made some slight modifications and added a link to the doc page that you shared. Let me know it looks good to go - thanks!

```suggestion
* Introduces <<endpoint-data-volume,new advanced settings>> in the {elastic-defend} integration policy to reduce the volume of data that {elastic-endpoint} processes and ingests. The following new behaviors are enabled by default. You can turn them off by configuring your {elastic-defend} integration policy advanced settings: 
+
NOTE: {elastic-endpoint} behavior is preserved on existing {elastic-defend} policies.
+
** {elastic-endpoint} will merge short lived process `create/terminate` events and `network connect/terminate` events so only a single document is produced. 
** {elastic-endpoint} will only include a small subset of data in the `host.*` fieldset in event documents.
** {elastic-endpoint} will not report MD5 and SHA-1 hashes in event data.
```

> * Allows you to opt into aggregation of network events to reduce CPU usage, I/O, and event sizes. Network events with the same addresses and ports occurring in rapid succession will be combined into fewer aggregate events. Use the `advanced.events.aggregate_network` advanced setting in your {elastic-defend} policy to enable it.

Reviewer 8: Looks good!

* Provides APIs for AI Assistant Knowledge Base entries ({kibana-pull}206407[#206407]).
* Adds the product documentation tool to AI Assistant to ensure product docs are installed and can be properly retrieved ({kibana-pull}199694[#199694]).
* Introduces support for the future integration of AI Assistant prompts in {kib}. ({kibana-pull}207138[#207138]).
* Adds audit logging for changes to AI Assistant knowledge base entries ({kibana-pull}203349[#203349]).
* Adds a service example to the entity store upload page ({kibana-pull}209023[#209023]).
* Update entity insight badge to open entity flyouts ({kibana-pull}208287[#208287]).
* Introduces changes to the entity analytics feature to support `event.ingested` as a configurable timestamp field for init and enable endpoints ({kibana-pull}208201[#208201]).
* Allows you to include closed alerts in risk score calculations ({kibana-pull}201909[#201909]).
* Turns the `securitySolution:enableVisualizationsInFlyout` <<visualizations-in-flyout,advanced setting>> on by default, which allows you to access the event analyzer and Session View in the **Visualize** tab on the alert or event details flyout ({kibana-pull}211319[#211319]).

> * Reduces the system performance impact of file events.

Reviewer 8: @gabriellandau is this for Elastic Defend?

* Improves {elastic-defend}'s resilience in low memory situations.
* Updates the {elastic-defend} status message ACK'ed to Agent to show: the {elastic-defend} policy name, revision, and Agent policy revision.
* Ensures that the data view selector on the rule creation form shows data view names instead of their defined indices ({kibana-pull}214495[#214495]).
* Implements various performance optimizations to reduce {elastic-defend}'s CPU usage and improve system responsiveness.
* Includes the {elastic-defend} policy name and ID in alerts.

> * Adds the `allow_cloud_features` advanced policy setting, which lets you explicitly list which cloud resources can be reached by {elastic-defend}.

Author: Just fixing up the link at the end:

```suggestion
* Adds the `allow_cloud_features` advanced policy setting, which lets you explicitly list which cloud resources can be reached by {elastic-defend} ({kibana-pull}205785[#205785]).
```

* Fixes a bug that prevented you from seeing alert assignee details from the Alerts table or the alert details flyout ({kibana-pull}211824[#211824]).
* Fixes the width of the alerts table in rule preview ({kibana-pull}214028[#214028]).
* Fixes a bug that prevented the rule creation form from properly validating EQL queries when you added filters to the query ({kibana-pull}212117[#212117]).
* Makes 7.x alert indices compatible with Alerts table so you can access alerts in legacy indices ({kibana-pull}209936[#209936]).
* Fixes a bug that didn't allow you to generate {esql} alerts from alert indices ({kibana-pull}208894[#208894]).
* Surfaces details for failed EQL non-sequence queries on the rule details page and in the event log ({kibana-pull}207396[#207396]).
* Fixes an {elastic-defend} bug where environment variables were not collected on macOS according to the `advanced.capture_env_vars` field.
* Fixes an {elastic-defend} bug to ensure the first event's timestamp is used as the timestamp for event aggregation.
* Updates the way {elastic-defend} initially connects to {agent}, which significantly improves the speed of connection.

> * Fixes issues where uninstalling Windows Defend leaves files within {elastic-defend}'s directory that cannot be removed by administrators. These leftover files can prevent subsequent installs and upgrades.

Reviewer 8: ```suggestion
* Fixes issues where uninstalling {elastic-defend] on Windows leaves files within {elastic-defend}'s directory that cannot be removed by administrators. These leftover files can prevent subsequent installs and upgrades.
```

[[release-notes-header-8.18.0]]
== 8.18

[discrete]
[[release-notes-8.18.0]]
=== 8.18.0

[discrete]

> [[known-issue-8.18.0]]

Reviewer 3: Apologies as this is last minute, but created a ticket about needing to doc [this](https://github.com/elastic/docs-content/issues/1021) as a known issue. Would it be possible to add here?

Author: Added in [9f38601](https://github.com/elastic/security-docs/pull/6519/commits/9f38601fdcaf44f9df30d82a31061fd06ea8c608).

* Allows you to https://github.com/elastic/kibana/issues/174168[customize prebuilt rules]. You can modify most rule parameters, export and import prebuilt rules — including customized ones — and upgrade prebuilt rules while retaining customization settings ({kibana-pull}212761[#212761]).
* Adds initial support for the service entity type in the Entity Store, whereas previously, only user and host entity types were supported ({kibana-pull}207336[#207336], {kibana-pull}206582[#206582], {kibana-pull}206268[#206268], {kibana-pull}202344[#202344]).
* Allows you to configure how often the enrich policy runs for the entity store ({kibana-pull}207374[#207374], {kibana-pull}204437[#204437]).
* Provides configuration options to the entity store through additional API parameters ({kibana-pull}206421[#206421]).
* Introduces a status tab to the entity store management page ({kibana-pull}201235[#201235]).
* Allows you to install and reinstall entity stores from the Engine Status page ({kibana-pull}208149[#208149]).
* Introduces ways to monitor and fix gaps in rule executions, which can lead to missed alerts or reduced rule coverage ({kibana-pull}206313[#206313]).

> * The manual runs functionality is now generally available ({kibana-pull}209535[#209535]).

Reviewer 9: Does it make sense to group the GA functions one after the other instead of lines apart?

Author: I don't have any issue with that but it'd break the current organization principle, which is to group changes based on product area.

+
** {elastic-endpoint} will merge short lived process `create/terminate` events and `network connect/terminate` events so only a single document is produced. 
** {elastic-endpoint} will only include a small subset of data in the `host.*` fieldset in event documents.
** {elastic-endpoint} will not report MD5 and SHA-1 hashes in event data.

[discrete]
[[enhancements-8.18.0]]
==== Enhancements
* Enhances Attack discovery by providing you with additional control over which alerts are included as context to the large language model (LLM) ({kibana-pull}205070[#205070]).
* Provides APIs for AI Assistant Knowledge Base entries ({kibana-pull}206407[#206407]).
* Adds the product documentation tool to AI Assistant to ensure product docs are installed and can be properly retrieved ({kibana-pull}199694[#199694]).
* Introduces support for the future integration of AI Assistant prompts in {kib}. ({kibana-pull}207138[#207138]).
* Adds audit logging for changes to AI Assistant knowledge base entries ({kibana-pull}203349[#203349]).
* Adds a service example to the entity store upload page ({kibana-pull}209023[#209023]).

> * Update entity insight badge to open entity flyouts ({kibana-pull}208287[#208287]).

Reviewer 9: ```suggestion
* Updates the entity insight badge to open entity flyouts ({kibana-pull}208287[#208287]).
```

* Improves {elastic-defend} script visibility and adds a new API event for `AmsiScanBuffer`, as well as AMSI enrichments for API events.
* Enhances {elastic-defend} by including an improved fingerprint for `Memory_protection.unique_key_v2`. We recommend that any `shellcode_thread` exceptions based on the old `unique_key_v1` field be updated.
* Adds the `process.Ext.memory_region.region_start_bytes` field to {elastic-defend} Windows memory signature alerts.
* Improves {elastic-defend} host information accuracy, such as IP addresses. {elastic-defend} was updating this information only during new policy application or at least once ever 24 hours, so this information could have been inaccurate for several hours, especially on roaming endpoints.

[discrete]
[[bug-fixes-8.18.0]]
==== Bug fixes
* Fixes the unstructured system log flow for Automatic Import ({kibana-pull}213042[#213042]).
* Fixes missing ECS mappings for Automatic Import ({kibana-pull}209057[#209057]).
* Fixes how Automatic Import generates accesses for the field names that are not valid Painless identifiers ({kibana-pull}205220[#205220]).

> * Automatic Import now ensures that the field mapping contains the `@timestamp` field whenever possible ({kibana-pull}204931[#204931]).

Reviewer 9: ```suggestion
* Ensures that the field mapping for Automatic Import contains the `@timestamp` field whenever possible ({kibana-pull}204931[#204931]).
```

Reviewer 10: ```suggestion
* Ensures that field mappings generated by Automatic Import contain the `@timestamp` field whenever possible ({kibana-pull}204931[#204931]).
```

* Improves the navigation and page descriptions for the Entity Store and Entity Risk Score pages ({kibana-pull}209130[#209130]).
* Fixes a bug that prevented the `indexPattern` parameter from being respected when you refreshed a data view ({kibana-pull}215151[#215151]). 
* Ensures that {kib} space IDs are dynamically retrieved for entity risk scores in the entity flyout ({kibana-pull}216063[#216063]). 
* Uses data from the risk engine's saved object instead of your browser's local storage when loading the Entity Risk Score page ({kibana-pull}215304[#215304]). 
* Improves the confirmation message that appears when you update the configuration for a risk engine saved object ({kibana-pull}211372[#211372]).
* Fixes a navigation issue with the host and user flyouts that prevented the flyout details from refreshing ({kibana-pull}209863[#209863]).
* Ensures that you stay on your current page in the Rules table after editing or updating a rule ({kibana-pull}209537[#209537]).
* Fixes a bug that caused the preview panel to incorrectly persist after you opened the session viewer preview ({kibana-pull}213455[#213455]).

> * Adds a "no data message" to the expanded event analyzer view in the alert details flyout when the event analyzer isn't turned on ({kibana-pull}211981[#211981]).

Reviewer 9: ```suggestion
* Adds a "no data" message to the expanded event analyzer view in the alert details flyout when the event analyzer isn't turned on ({kibana-pull}211981[#211981]).
```

[[release-notes-header-8.18.0]]
== 8.18

[discrete]
[[release-notes-8.18.0]]
=== 8.18.0

[discrete]
[[known-issue-8.18.0]]
==== Known issues
// tag::known-issue[]
[discrete]
.Rules cannot be enabled if they're corrupted while upgrading from 7.17.x to 8.x 
[%collapsible]
====
*Details* +
If rule saved objects were corrupted when you upgraded from 7.17.x to 8.x, you may run into an error when turning on your rules. 

*Workaround* +

Duplicate your rules and enable them.

====
// end::known-issue[]

// tag::known-issue[]

> [discrete]

Reviewer 3: Thank you!

.The technical preview badge incorrectly displays on the alert suppression fields for event correlation rules 
[%collapsible]
====
*Details* +

> Alert suppression for event correlation rules is generally available but still shows as being in technical preview when you create a new rule.  

Author: ```suggestion
On April 8, 2025, it was discovered that alert suppression for event correlation rules is incorrectly shown as being in technical preview when you create a new rule. For more information, check (https://github.com/elastic/docs-content/issues/1021)[#1021].
```

Reviewer 4: How was this discovered in the future (tomorrow the 8th)? Perhaps you meant April 3rd?
